src/Sonarr.Http/REST/RestController.cs

```diff
@@ -70,6 +70,12 @@ public override void OnActionExecuting(ActionExecutingContext context)
 
                 foreach (var resource in resourceArgs)
                 {
+                    // Map route Id to body resource if not set in request
+                    if (Request.Method == "PUT" && resource.Id == 0 && context.RouteData.Values.TryGetValue("id", out var routeId))
+                    {
+                        resource.Id = int.Parse(routeId.ToString());
```

Comment:
```suggestion
                        resource.Id = Conver.ToInt32(routeId);
```

This avoids the warning with converting to string if `routeId` is null, but otherwise functions the same.

Reply:
Looks good. This one actually effects downstream as well, so good catch on the issue OP